Ack connect route (#2780)
aws-samples/serverless-patterns: ack-connect-route/README.md

> ## Websocket API Gateway acknowledgement for $connect route.

Reviewer: ```suggestion
## Websocket acknowledgement for $connect route in Amazon API Gateway
```

> The Serverless Application Model (SAM) template deploys an Amazon WebSocket API Gateway and two AWS Lambda functions. When a client connects, the API Gateway creates a $connect route with Lambda proxy integration. The first Lambda function processes the initial connection, capturing both the Connection ID and API Gateway stage URL, then asynchronously triggers the second Lambda function. This second function validates the Connection ID and, if valid, uses SDK API calls to send a greeting message back to the client.

Reviewer: ```suggestion
The Serverless Application Model (SAM) template deploys an Amazon WebSocket API Gateway and two AWS Lambda functions. When a client connects, the API Gateway creates a $connect route with Lambda proxy integration. The first Lambda function processes the initial connection, capturing both the Connection ID and API Gateway stage URL, then asynchronously invokes the second Lambda function. This second function validates the Connection ID and, if valid, uses SDK API calls to send a greeting message back to the client.
```

* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed

## Deployment Instructions

1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
    ``` 
    git clone https://github.com/aws-samples/serverless-patterns
    ```
2. Change directory to the pattern directory:
    ```
    cd ack-connect-route
    ```
3. From the command line, use AWS SAM to build and deploy the AWS resources for the pattern as specified in the template.yml file:

    ```
    sam deploy --guided
    ```
4. During the prompts:
    * Enter a stack name
    * Enter the desired AWS Region
    * Allow SAM CLI to create IAM roles with the required permissions.
          

>          Once you have run guided mode once, you can use `sam deploy` in future to use these defaults.

Reviewer: ```suggestion
         After running guided mode once, you can use `sam deploy` in future to use these defaults.
```

1. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.

## Testing

> Once the application is deployed, retrieve the WebSocketURL value from CloudFormation Outputs. To test the WebSocket API, you can use [wscat](https://github.com/websockets/wscat) which is an open-source command line tool.

Reviewer: ```suggestion
After deployment, retrieve the `WebSocketURL` value from CloudFormation Outputs. To test the WebSocket API, you can use [wscat](https://github.com/websockets/wscat) which is an open-source command line tool.
```

1. [Install NPM](https://www.npmjs.com/get-npm).

2. Install wscat:
    ```
    $ npm install -g wscat
    ```

3. Connect to your WebSocketURL by executing the following command:
    $ wscat -c <YOUR WEBSOCKET URL>
    ```

> 4. To test the custom route and its associated function, send a JSON-formatted request. The Lambda function sends back the value of the "data" key using the callback URL:

Reviewer: Provide an example of the JSON formatted event.

Author: done

```
$ wscat -c <YOUR WEBSOCKET URL>
connected (press CTRL+C to quit)
```
## Cleanup

1. Delete the stack
    ```

>     aws cloudformation delete-stack --stack-name <YOUR STACK NAME>

Reviewer: Use `sam delete` since you are using SAM
